server/src/main/java/org/opensearch/cluster/service/applicationtemplates/SystemTemplatesService.java

```
@@ -0,0 +1,117 @@
+
+    public static final Setting<Integer> SETTING_APPLICATION_BASED_CONFIGURATION_TEMPLATES_LOAD = Setting.intSetting(
+        "cluster.application_templates.load",
+        APPLICATION_BASED_CONFIGURATION_TEMPLATES_LOAD_DEFAULT_COUNT,
+        0,
+        1000,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    private final List<SystemTemplatesPlugin> systemTemplatesPluginList;
+
+    private final Settings settings;
+
+    private final AtomicBoolean loaded = new AtomicBoolean(false);
+
+    private volatile int templatesToLoad = APPLICATION_BASED_CONFIGURATION_TEMPLATES_LOAD_DEFAULT_COUNT;
+
+    private static final Logger logger = LogManager.getLogger(SystemTemplatesService.class);
+
+    public SystemTemplatesService(List<SystemTemplatesPlugin> systemTemplatesPluginList,
+                                  ClusterSettings clusterSettings,
+                                  Settings settings) {
+        this.systemTemplatesPluginList = systemTemplatesPluginList;
+        clusterSettings.addSettingsUpdateConsumer(SETTING_APPLICATION_BASED_CONFIGURATION_TEMPLATES_LOAD, this::setTemplatesToLoad);
+        this.settings = settings;
+    }
+
+    @Override
+    public void onClusterManager() {
+        refreshTemplates();
+    }
+
+    @Override
+    public void offClusterManager() {
+        // do nothing
+    }
+
+    private void refreshTemplates() {
+        if (loaded.compareAndSet(false, true)) {
+            if (SETTING_APPLICATION_BASED_CONFIGURATION_TEMPLATES_LOAD.get(settings) > 0) {
+                int countOfTemplatesToLoad = templatesToLoad;
+                int templatesLoaded = 0;
+                int failedLoadingTemplates = 0;
+                int failedLoadingRepositories = 0;
+
+                for (SystemTemplatesPlugin plugin : systemTemplatesPluginList) {
+                    try (TemplateRepository repository = plugin.loadRepository()) {
```

SystemTemplatesRepository is not mandatory? as we swallowing the failures

---

Since there can be multiple repositories and multiple templates, I was passing forward by ignoring failures due to loading of 1 repo and move on to the next.